### PR TITLE
Epytope switch in process peptide_prediction

### PR DESCRIPTION
**Pull request by jonasscheid**
The new release of epytope (v3.0.0) provides an altered prediction structure due to the incorporation of rank metrics.
This PR aims to embed the new prediction structure in the `epaa.py` script, which is called by the `PEPTIDE_PREDICTION` process.

Related issues:
#54 
#130

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/epitopeprediction _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by christopher-mohr** [comment]
I will have a look. 👍 
Could you also please change conda and the containers in the process [here](https://github.com/nf-core/epitopeprediction/blob/8bb383c44383304c69557b4d00366384195918d5/modules/local/peptide_prediction.nf#L4). Should be `bioconda::epytope` and `quay.io/biocontainers/epytope`. I think you might have to specify the `tag` as well. If it works the test should pass as well.

**Comment by christopher-mohr** [comment]
There was a permission problem with the bioconda container which is fixed now. The container can now be pulled from `http://quay.io/biocontainers/epytope:3.0.0--pyh5e36f6f_0`.

File: bin/epaa.py [hunk 1]
```diff
@@ -694,38 +688,18 @@ def get_protein_ids_for_transcripts(idtype, transcripts, ensembl_url, reference)
...
-    if not pd.isnull(j):
-        if 'syf' in method:
-            return round(j, 4)
-        elif any(m in method for m in ['mhcnuggets','mhcflurry']):
-            # convert IC50 -> affinity score in [0,1]
-            min_ic50 = 1.0
-            return round(1.0 - math.log(max(j,min_ic50), 50000), 4)
-        else:
-            # currently not applied (for methods such as netMHC)
-            return round(j, 4)
```
Comment (christopher-mohr) [inline bin/epaa.py:713]: might make sense to keep the `round` for the scores column to get the same results as before.

File: bin/epaa.py [hunk 2]
```diff
@@ -821,44 +795,57 @@ def make_predictions_from_variants(variants_all, methods, tool_thresholds, allel
...
+
+        # merge remaining multi-column Allele and ScoreType
+        df.columns = df.columns.map('{0[0]} {0[1]}'.format)
+
+        # reset index to have indices as columns
+        df.reset_index(inplace=True)
+        df = df.rename(columns={'Method': 'method', 'Peptides':'sequence'})
+
+        # create column containing the peptide lengths
+        #df.insert(2, 'length', df['sequence'].map(len))
```
Comment (christopher-mohr) [inline bin/epaa.py:818]: can be removed I guess?

**Review by christopher-mohr — CHANGES_REQUESTED** [review]
Some things that need to be changed but overall it looks good. 👍

File: bin/epaa.py [hunk 3]
```diff
@@ -694,38 +688,18 @@ def get_protein_ids_for_transcripts(idtype, transcripts, ensembl_url, reference)
...
     try:
-        pssm = getattr(__import__("Fred2.Data.pssms.syfpeithi.mat."+allele_model, fromlist=[allele_model]), allele_model)
-        return sum([max(scrs.values()) for pos, scrs in pssm.iteritems()])
+        pssm = getattr(__import__("epytope.Data.pssms.syfpeithi.mat."+allele_model, fromlist=[allele_model]), allele_model)
+        return sum([max(scrs.values()) for pos, scrs in pssm.items()])
     except:
         return np.nan
 
 
-def create_score_values(j, method):
```
[the remark this answers is not in the document]
Comment (jonasscheid) [reply bin/epaa.py:703]: As discussed, epytope returns score values for all predictors. Therefore, the method is not needed anymore. Values are rounded in a different step

File: bin/epaa.py [hunk 4]
```diff
@@ -322,9 +322,6 @@ def read_vcf(filename, pass_only=True):
 
                     for sample in record.samples:
                         for format_key in format_list:
-                            if not getattr(sample.data, format_key, None):
```
[the remark this answers is not in the document]
Comment (jonasscheid) [reply bin/epaa.py:325]: Yep, will add it again.

File: bin/epaa.py [hunk 5]
```diff
@@ -694,38 +688,18 @@ def get_protein_ids_for_transcripts(idtype, transcripts, ensembl_url, reference)
...
-
-    else:
-        return np.nan
-
-
 def create_affinity_values(allele, length, j, method, max_scores, allele_strings):
     if not pd.isnull(j):
         if 'syf' in method:
             return max(0, round((100.0 / float(max_scores[allele_strings[('%s_%s' % (str(allele), length))]]) * float(j)), 2))
-        elif any(m in method for m in ['mhcnuggets','mhcflurry']):
```
[the remark this answers is not in the document]
Comment (jonasscheid) [reply bin/epaa.py:723]: See above

**Review by christopher-mohr — APPROVED** [review]
LGTM! Some of the tests are failing because of an `epytope` issue which will be solved in the new release.

**Pull request merged**